docs/contracts.rst

```diff
@@ -1,8 +1,72 @@
+.. _contracts:
+
 Contracts
 =========
 
 .. py:module:: web3.contract
 
+It is worth taking your time to understand all about contracts. To get started,
+check out this example:
+
+.. _contract_example:
+
+Simple Contract Example
+-----------------------
+
+.. code-block:: python
+
+    import json
+    import web3
+
+    from web3 import Web3, TestRPCProvider
+    from solc import compile_source
+    from web3.contract import ConciseContract
+
+    # Solidity source code
+    contract_source_code = '''
+    pragma solidity ^0.4.0;
+
+    contract Greeter {
+        string public greeting;
+
+        function Greeter() {
+            greeting = 'Hello';
+        }
+
+        function setGreeting(string _greeting) public {
+            greeting = _greeting;
+        }
+
+        function greet() constant returns (string) {
+            return greeting;
+        }
+    }
+    '''
+
+    compiled_sol = compile_source(contract_source_code) # Compiled source code
+    contract_interface = compiled_sol['<stdin>:Greeter']
+
+    # web3.py instance
+    w3 = Web3(TestRPCProvider())
+
+    # Instantiate and deploy contract
+    contract = w3.eth.contract(abi=contract_interface['abi'], bytecode=contract_interface['bin'])
+
+    # Get transaction hash from deployed contract
+    tx_hash = contract.deploy(transaction={'from': w3.eth.accounts[0], 'gas': 410000})
+
+    # Get tx receipt to get contract address
+    tx_receipt = w3.eth.getTransactionReceipt(tx_hash)
```

Comment:
May be use `waitForTransactionReceipt` instead? I think it would be better to give an example that would work on the main network too.